crates/executor/src/swarm.rs

```diff
@@ -0,0 +1,137 @@
+use futures::StreamExt;
+use libp2p::gossipsub::{self, IdentTopic};
+use libp2p::identity::Keypair;
+use libp2p::swarm::{NetworkBehaviour, SwarmEvent};
+use libp2p::{mdns, noise, tcp, yamux, SwarmBuilder};
+use std::error::Error;
+use std::time::Duration;
+use tokio::sync::mpsc;
+use tokio::task::JoinHandle;
+use tracing::{debug, error};
+use zetina_common::graceful_shutdown::shutdown_signal;
+
+#[derive(NetworkBehaviour)]
+pub struct PeerBehaviour {
+    gossipsub: gossipsub::Behaviour,
+    mdns: mdns::tokio::Behaviour,
+}
+
+pub struct SwarmRunner {
+    handle: Option<JoinHandle<()>>,
+}
+
+impl SwarmRunner {
+    pub fn new(
+        p2p_local_keypair: &Keypair,
+        subscribe_topics: Vec<IdentTopic>,
+        mut transmit_topics: Vec<(IdentTopic, mpsc::Receiver<Vec<u8>>)>,
+        swarm_events_tx: mpsc::Sender<gossipsub::Event>,
```

Comment:
same as above on delegator SwarmRunner side. i'd like to just use common object for both